File: src/types/events.ts
export interface Event {
  event_id: string;
  bus_id: string;
  timestamp: string;
  latitude: number;
  longitude: number;
  road_segment_id: string;
  event_type: 'road_defect' | 'waterlogging' | 'traffic' | 'incident';
  class_name?: string;
  confidence: number;
  severity?: number;
  severity_label?: string;
  frame_id?: number;
  evidence_uri?: string;
  risk_score?: number;
  risk_level?: string;
  breadth_cm?: number;
  depth_cm?: number;
  dimensions?: {
    breadth_cm?: number;
    depth_cm?: number;
    [key: string]: any;
  };
  risk_assessment?: string;
  metadata?: Record<string, any>;
}

File: backend/app/schemas/events.py
from typing import Optional, Literal
from pydantic import BaseModel, Field


class EventResponse(BaseModel):
    event_id: str
    bus_id: str
    timestamp: str
    latitude: float
    longitude: float
    road_segment_id: str
    event_type: Literal["road_defect", "waterlogging", "traffic", "incident"]
    class_name: Optional[str] = None
    confidence: float
    severity: int = Field(..., ge=1, le=4, description="Numeric severity: 1 (Low), 2 (Moderate), 3 (High), 4 (Critical)")
    severity_label: Optional[str] = Field(default=None, description="Descriptive severity label for display only: low, moderate, high, critical")
    frame_id: Optional[int] = None
    evidence_uri: Optional[str] = None
